e2e/tests/pages/hearing.spec.js

```diff
@@ -0,0 +1,119 @@
+  });
+
+  test('should display organization', async () => {
+    await expect(page.locator('main')).toContainText(hearing.organization);
+  });
+
+  test('should have comment link', async () => {
+    const link = page.locator('.internal-link:has-text("Kirjoita kommentti")');
+    await expect(link).toBeVisible();
+  });
+
+  test('should display comment count', async () => {
+    await expect(page.getByTestId('comment-summary')).toContainText(`Yhteensä ${hearing.n_comments} kommenttia`);
+  });
+
+  test('should have Swedish link if available', async () => {
+    const noSwedishTitle = !hearing.title.sv;
+    test.skip(noSwedishTitle, 'No Swedish title available');
+
+    await expect(page.getByTestId('language-select')).toContainText('Hörandet tillgängligt på svenska');
+  });
+
+  test('should have English link if available', async () => {
+    const noEnglishTitle = !hearing.title.en;
+    test.skip(noEnglishTitle, 'No English title available');
+
+    await expect(page.getByTestId('language-select')).toContainText('Hearing available in English');
+  });
+
+  test('should display label', async () => {
+    const firstFinnishLabel = hearing.labels[0].label.fi;
+
+    if (firstFinnishLabel) {
+      await expect(page.locator('[aria-label="Asiasanat"]')).toContainText(firstFinnishLabel);
+    }
+  });
+
+  test('should display contact person name', async () => {
+    const contactPerson = hearing.contact_persons[0];
+    await expect(page.locator('.hearing-contacts')).toContainText(contactPerson.name);
+  });
+
+  test('should have image with caption', async () => {
+    const img = page.locator('.section-image img');
+    await expect(img).toHaveAttribute('alt', hearing.main_image.caption.fi);
+  });
+
+  test('should have Excel link for comment summary', async () => {
+    const link = page.locator('a:has-text("Lataa yhteenveto kommenteista Excel-muodossa")');
+    await expect(link).toBeVisible();
+  });
+
+  test('should have vote and comment button', async () => {
+    const button = page.locator('button:has-text("Äänestä ja kommentoi")').first();
+    await expect(button).toBeVisible();
+  });
+
+  test('verify visibility of questions and their options', async () => {
+    await page.getByRole('button', { name: 'Äänestä ja kommentoi' }).last().click();
+
+    const mainSection = hearing.sections.find((s) => s.type === 'main');
+    test.skip(!mainSection, 'No main section found');
+    const firstQuestion = mainSection.questions[0];
+    const questionText = firstQuestion.text.fi;
+    const options = firstQuestion.options.map((o) => o.text.fi);
+    await expect(page.locator('.comment-form')).toContainText(questionText);
+    await expect(page.getByTestId('question-form-group').locator('.radio')).toContainText(options);
+  });
+
+  test('user can successfully submit a comment', async () => {
+    await page.getByRole('button', { name: 'Äänestä ja kommentoi' }).last().click();
```

Comment:
I think there is no guarantee that hearings always have voting enabled, so probably should make this more conditional. 🤔